test/functional/test_framework/wallet.py

```diff
@@ -64,16 +73,28 @@ def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_sp
         fee = utxo_to_spend['value'] - send_value
         assert send_value > 0
 
-        tx = CTransaction()
-        tx.vin = [CTxIn(COutPoint(int(utxo_to_spend['txid'], 16), utxo_to_spend['vout']))]
-        tx.vout = [CTxOut(int(send_value * COIN), self._scriptPubKey)]
-        tx.wit.vtxinwit = [CTxInWitness()]
-        tx.wit.vtxinwit[0].scriptWitness.stack = [CScript([OP_TRUE])]
+        tx = self.prepare_tx(utxo_to_spend, send_value)
         tx_hex = tx.serialize().hex()
-
         txid = from_node.sendrawtransaction(tx_hex)
         self._utxos.append({'txid': txid, 'vout': 0, 'value': send_value})
         tx_info = from_node.getmempoolentry(txid)
         assert_equal(tx_info['vsize'], vsize)
         assert_equal(tx_info['fee'], fee)
         return {'txid': txid, 'wtxid': tx_info['wtxid'], 'hex': tx_hex}
+
+    def send_big_self_transfer(self, *, fee, from_node, utxo_to_spend=None, big_txouts):
```

Comment:
I think `big_txouts` should come before the optional `utxo_to_spend=None` to spend now that it's required, right?